experimental/Matrix/matrix.g

```diff
@@ -0,0 +1,195 @@
+        if k < 0 then
+            return MakeJuliaMatrixRep( (Inverse(m1)!.m)^(-k) );
+        else
+            return MakeJuliaMatrixRep( (m1!.m)^k );
+        fi;
+    end);
+    
+InstallOtherMethod(Display, [IsJuliaMatrixRep], function( m )
+    local i,j;
+        Print("[ ");
+        for i in [1..NumberRows(m)] do
+            Print("[ ");
+            for j in [1..NumberColumns(m)] do
+                Print(m[i,j]);
+                if (j < NumberColumns(m)) then
+                    Print(", ");
+                fi;
+            od;
+            if (i < NumberRows(m)) then
+                Print(" ] \n");
+                Print("  ");
+            else
+                Print(" ] ] \n");
+            fi;
+        od;
+    end);
+
+InstallOtherMethod(Unpack, [IsJuliaMatrixRep], function(m)
+    local v, i, j;
+    v := EmptyPlist(NumberRows(m)*NumberColumns(m));
+    for i in [1..NumberRows(m)] do
+        for j in [1..NumberColumns(m)] do
+            v[j+NumberColumns(m)*(i-1)] := m[i,j];
+        od;
+    od;
+    return v;
+end);
+
+
+# Method: SetNiceMorphismForJuliaMatrixRepGroup
+# Input: Matrix group G
+# Output: -
+# This function sets a nice morphism for G. The morphism is defined from the Julia matrix to the corresponding GAP matrix
+
+DeclareOperation("SetNiceMorphismForJuliaMatrixRepGroup", [IsGroup]);
+
+InstallMethod(SetNiceMorphismForJuliaMatrixRepGroup, [IsGroup], function(G)
+    local hom, f, f_inv, ele, GAPGenerators, i, gens, GAPGroup, JuliaGAPMap;
+    
+        gens := GeneratorsOfGroup(G);
+        Assert(0, ForAll(gens, IsJuliaMatrixRep));
+        
+        ele := gens[1];
+        hom := Julia.Oscar._iso_oscar_gap(ele!.m.base_ring);
+        
+        GAPGenerators := List(gens, g -> Julia.AbstractAlgebra.map_entries(hom, g!.m));
+        
+        GAPGroup := GroupByGenerators(GAPGenerators);
+        
+        f := function(m) return Julia.AbstractAlgebra.map_entries(hom,m!.m); end;
+        f_inv := function(m) return MakeJuliaMatrixRep(Julia.Oscar.preimage_matrix(hom,m)); end;
+
+        JuliaGAPMap := GroupHomomorphismByFunction(G,GAPGroup,f,f_inv);
+        SetNiceMonomorphism(G,JuliaGAPMap);
+        SetIsHandledByNiceMonomorphism(G, true);
+    end);
+    
+InstallOtherMethod(\=, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        return m1!.m = m2!.m;
+    end);
+    
+InstallOtherMethod(\<, [IsJuliaMatrixRep, IsJuliaMatrixRep], function( m1, m2 )
+        if Julia.applicable(Julia.Oscar.MatrixGroups._lex_isless, m1!.m, m2!.m ) then
+            return Julia.Oscar.MatrixGroups._lex_isless(m1!.m,m2!.m);
+        fi;
+        if Julia.applicable(Julia.isless,m,m1) then
+            return m1!.m < m2!.m;
+        fi;
+        Error("comparing IsJuliaMatrixRep not supported");
+    end);
+
+InstallOtherMethod(MinimalPolynomial, [IsJuliaMatrixRep], function(m)
+    return Julia.AbstractAlgebra.minpoly(m!.m);
```

Comment:
I would suggest to reference `Oscar` instead of `AbstractAlgebra`, to be robust against possible future changes
```suggestion
    return Julia.Oscar.minpoly(m!.m);
```